### PR TITLE
[CI] Allow to skip all tests in the parallel implementation.

### PR DESCRIPTION
**Pull request by mandel-macaque**
Since we are generating jobs in parallel, we want to not even create the jobs which is more efficient than letting xharness skip the tests.

### CODE REVIEW

**Review by dalexsoto — APPROVED** [review]

File: tools/devops/automation/templates/tests/stage.yml [hunk 1]
```diff
@@ -103,6 +103,8 @@ stages:
 
   - ${{ each label in parameters.simTestsConfigurations }}:
     - job: "tests_${{ replace(label, '-', '_') }}"
+      ${{ if eq(parameters.parseLabels, true) }}:
+        condition: eq(dependencies.configure.outputs['labels.build-package'], '')
```
Comment (rolfbjarne) [inline tools/devops/automation/templates/tests/stage.yml:107]: Shouldn't this be:

```suggestion
        condition: eq(dependencies.configure.outputs['labels.skip-all-tests'], '')
```

File: tools/devops/automation/templates/tests/stage.yml [hunk 2]
```diff
@@ -144,13 +146,18 @@ stages:
           gitHubToken: ${{ parameters.gitHubToken }}
           xqaCertPass: ${{ parameters.xqaCertPass }}
 
-  - job: publish_html
-    displayName: 'Publish Html report in VSDrops'
+  - job: publish_test_results
+    ${{ if eq(parameters.parseLabels, true) }}:
+      condition: and(eq(dependencies.configure.outputs['labels.build-package'], ''),  succeededOrFailed())
```
Comment (rolfbjarne) [inline tools/devops/automation/templates/tests/stage.yml:151]:
```suggestion
      condition: and(eq(dependencies.configure.outputs['labels.skip-all-tests'], ''),  succeededOrFailed())
```

File: tools/devops/automation/templates/tests/stage.yml [hunk 3]
```diff
@@ -169,3 +176,34 @@ stages:
         statusContext: ${{ parameters.statusContext }}
         vsdropsPrefix: ${{ parameters.vsdropsPrefix }}
         testPrefix: ${{ parameters.testPrefix }}
+
+  - job: publish_skipped_tests
+    ${{ if eq(parameters.parseLabels, true) }}:
+      condition: ne(dependencies.configure.outputs['labels.build-package'], '')
```
Comment (rolfbjarne) [inline tools/devops/automation/templates/tests/stage.yml:182]:
```suggestion
      condition: ne(dependencies.configure.outputs['labels.skip-all-tests'], '')
```

File: tools/devops/automation/templates/tests/stage.yml [hunk 4]
```diff
@@ -169,3 +176,34 @@ stages:
...
+    ${{ else }}:
+      condition: false
+    displayName: 'GitHub comment - Skipped tests'
+    timeoutInMinutes: 1000
+    dependsOn: # has to wait for the tests to be done AND the data to be uploaded
+    - ${{ if eq(parameters.parseLabels, true) }}:
+      - configure
```
Comment (rolfbjarne) [inline tools/devops/automation/templates/tests/stage.yml:189]: Is the condition here necessary?

```suggestion
    -  configure
```

File: tools/devops/automation/templates/tests/stage.yml [hunk 5]
```diff
@@ -169,3 +176,34 @@ stages:
...
+        clean: all
+
+    steps:
+    - pwsh: |
+        Import-Module $Env:SYSTEM_DEFAULTWORKINGDIRECTORY\xamarin-macios\tools\devops\automation\scripts\MaciosCI.psd1
+
+        $gihubComments = New-GitHubCommentsObject -Org "xamarin" -Repo "xamarin-macios" -Token $Env:GITHUB_TOKEN -Hash $Env:GIT_HASH -Debug
+        $sb = [System.Text.StringBuilder]::new()
+        $sb.AppendLine("All tests have been skipped because the label 'skip-all-tests' was set.")
+        $result = $gihubComments.NewCommentFromMessage("Test results", ":bangbang:", $sb.ToString())
```
Comment (rolfbjarne) [inline tools/devops/automation/templates/tests/stage.yml:203]: :bangbang: this is all good, so it seems like we should have a green emojii instead? :seedling:

```suggestion
        $result = $gihubComments.NewCommentFromMessage("Test results", ":seedling:", $sb.ToString())
```

**Review by rolfbjarne — CHANGES_REQUESTED** [review]

File: tools/devops/automation/templates/tests/stage.yml (hunk 4 above)
Comment (mandel-macaque) [reply tools/devops/automation/templates/tests/stage.yml:189]: Does make no much diff, less lines is better anyway.

File: tools/devops/automation/templates/tests/stage.yml [hunk 6]
```diff
@@ -144,13 +146,18 @@ stages:
...
           xqaCertPass: ${{ parameters.xqaCertPass }}
 
-  - job: publish_html
-    displayName: 'Publish Html report in VSDrops'
+  - job: publish_test_results
+    ${{ if eq(parameters.parseLabels, true) }}:
+      condition: and(eq(dependencies.configure.outputs['labels.skip-all-tests'], ''),  succeededOrFailed())
+    ${{ else }}:
+      condition: succeededOrFailed()
+    displayName: 'GiHub comment - Publish results'
```
Comment (rolfbjarne) [inline tools/devops/automation/templates/tests/stage.yml:154]:
```suggestion
    displayName: 'GitHub comment - Publish results'
```

File: tools/devops/automation/templates/tests/stage.yml [hunk 7]
```diff
@@ -144,13 +146,18 @@ stages:
...
     timeoutInMinutes: 1000
     dependsOn: # has to wait for the tests to be done AND the data to be uploaded
+    - ${{ if eq(parameters.parseLabels, true) }}:
+      - configure
     - ${{ each label in parameters.simTestsConfigurations }}:
       - tests_${{ replace(label, '-', '_') }}
-    condition: succeededOrFailed()
```
Comment (rolfbjarne) [inline tools/devops/automation/templates/tests/stage.yml:153]: Was this removal intentional? It seems we won't get the GH comment in case of failures now.

**Review by rolfbjarne — CHANGES_REQUESTED** [review]

**Pull request merged**